sdk/platform/http_client/curl/src/curl_client.cpp

```diff
@@ -32,5 +32,117 @@ Response CurlClient::send()
     }
   }
 
-  return Response(200, "OK\n");
+  return this->m_response;
+}
+
+CURLcode CurlClient::Perform()
+{
+  m_firstHeader = true;
+
+  auto settingUp = SetUrl();
+  if (settingUp != CURLE_OK)
+  {
+    return settingUp;
+  }
+
+  settingUp = SetHeaders();
+  if (settingUp != CURLE_OK)
+  {
+    return settingUp;
+  }
+
+  settingUp = SetWriteResponse();
+  if (settingUp != CURLE_OK)
+  {
+    return settingUp;
+  }
+
+  return curl_easy_perform(m_pCurl);
+}
+
+static void ParseAndSetFirstHeader(std::string const& header, Response** response)
+{
+  // set response code, http version and reason phrase (i.e. HTTP/1.1 200 OK)
+  auto start = std::find(header.begin(), header.end(), '/');
+  start++; // skip symbol
+  auto end = std::find(start, header.end(), '.');
```

Comment:
Ditto verification required, throughout. This function is very scary security wise so we should try very hard to reuse curl's parsing if it has that.

Reply:
by the time curl is calling our callback function with a header, curl has already made its validation to the response.
It will not call this function callback if server returned an invalid http response, instead, perfom() method will return some error and this call backs are not called